cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/queue/hbase/HBaseQueueClientFactory.java

```diff
@@ -229,6 +242,15 @@ private HTable createHTable(TableId tableId) throws IOException {
     return consumerTable;
   }
 
+  private int getDistributorBuckets(HTableDescriptor htd) {
+    String value = htd.getValue(QueueConstants.DISTRIBUTOR_BUCKETS);
+    // If the setting is not in the table meta, this is a old table, hence use the value in the cConf
+    if (value == null) {
+      return cConf.getInt(QueueConstants.ConfigKeys.QUEUE_TABLE_PRESPLITS);
```

Comment:
hmmm interesting. So if I have a 2.8 queue where the descriptor does not carry the DISTRIBUTOR_BUCKETS, then upgrade to 3.0, and change my cdap-site and set QUEUE_TABLE_PRESPLITS to 24, what happens? We would use 24 although the queue was created with 16 (the old default). Would it be more correct to say, if it is not in the HTable descriptor, then it must be a pre-3.0 queue and therefore it is 16?